snuba/consumers/consumer_builder.py

```diff
@@ -62,6 +65,39 @@ class MockParameters:
     std_deviation: int
 
 
+def validate_passed_slice(
+    storage_key: StorageKey, slice_id: Optional[int] = None
+) -> None:
+    # make sure this is a storage that is sliced
+    # and that the slice id passed in is valid
+    if slice_id is not None:
+        assert storage_key.value in SLICED_STORAGES
+        assert slice_id < SLICED_STORAGES[storage_key.value]
+
+
+def get_sliced_snuba_topic(
+    topic: SnubaTopic, slice_id: Optional[int] = None
+) -> SnubaTopic:
+    if slice_id is not None:
+        physical_topic = SLICED_KAFKA_TOPIC_MAP[(topic.value, slice_id)]
+        # this will require registration of the physical topic
+        return register_topic(physical_topic)
+
+    return topic
+
+
+def get_sliced_arroyo_topic(
+    passed_topic: str, slice_id: Optional[int] = None
+) -> ArroyoTopic:
+    if slice_id is not None:
+        physical_topic = SLICED_KAFKA_TOPIC_MAP[(passed_topic, slice_id)]
+        set_topic = ArroyoTopic(physical_topic)
+    else:
+        set_topic = ArroyoTopic(passed_topic)
+
+    return set_topic
```

Comment:
`KafkaTopicSpec` currently only works with Snuba topics. This PR has been updated to only include `get_physical_arroyo_topic()` (the Snuba equivalent was removed). Are you suggesting to make `KafkaTopicSpec` compatible with both topic types? i.e.something like:

```class KafkaTopicSpec:
    def __init__(self, snuba_topic: SnubaTopic, arroyo_topic: ArroyoTopic) -> None:
        self.__snuba_topic = snuba_topic
        self.__arroyo_topic = arroyo_topic
```

Reply:
It's not so much about SnubaTopic vs ArroyoTopic, more about logical vs physical topic and how it gets resolved it a common way.`KafkaTopicSpec` is supposed to be the singular way of resolving a Snuba logical topic name to a physical topic, via `KafkaTopicSpec.topic_name`. This is useful so that all code in Snuba can just reference the Snuba topic and get resolved in this exact same way and we don't need separate resolving functions like this in many parts of our codebase. This seems like it solves a similar problem as you are trying to do here so I would prefer we find a way to adapt that to make it work for sliced topics as well.

Reply:
I updated the PR to use `KafkaTopicSpec` for resolving the logical to sliced physical topic name